sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala

```diff
@@ -1233,6 +1242,34 @@ private[client] class Shim_v2_1 extends Shim_v2_0 {
   override def alterPartitions(hive: Hive, tableName: String, newParts: JList[Partition]): Unit = {
     alterPartitionsMethod.invoke(hive, tableName, newParts, environmentContextInAlterTable)
   }
+
+  override def getPartitionsByFilter(
+      hive: Hive,
+      table: Table,
+      predicates: Seq[Expression]): Seq[Partition] = {
+
+    // Hive getPartitionsByFilter() takes a string that represents partition
+    // predicates like "str_key=\"value\" and int_key=1 ..."
+    val filter = convertFilters(table, predicates)
+
+    val limit = SQLConf.get.metastorePartitionLimit
+    if (limit > -1) {
+      val num = try {
+        getNumPartitionsByFilterMethod.invoke(hive, table, filter).asInstanceOf[Int]
+      } catch {
+        case ex: Exception =>
+          logWarning("Caught Hive MetaException attempting to get partition metadata by " +
```

Comment:
Could you make the message clearer? e.g., ...get the number of partitions from ...., but .... using 0 for a partition number....